index.js

```diff
@@ -70,9 +70,9 @@ function inject (dispatchFunc, options, callback) {
 
     return req.prepare(() => dispatchFunc.call(server, req, res))
   } else {
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       const req = new Request(options)
-      const res = new Response(req, resolve)
+      const res = new Response(req, resolve, reject)
```

Comment:
I'd like to suggest doing something like this:

```js
const res = new Response(req, (err, res) => {
  if (err) reject(err)
  else resolve(res)
})
```

That way you wouldn't need the `if (this._promiseCallback)` checks in `lib/response.js` so `onEnd` could always just be a regular callback.

Up to you though :)

Reply:
Pr are welcome :)